engine/executor/dm/worker.go

```diff
@@ -318,3 +330,27 @@ func (w *dmWorker) checkAndAutoResume(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (w *dmWorker) discardResource4Syncer(ctx context.Context) {
+	if w.workerType != frameModel.WorkerDMSync || !w.needExtStorage {
+		return
+	}
+	impl, ok := w.unitHolder.(*unitHolderImpl)
+	if !ok {
+		return
+	}
+	isFresh, err := impl.unit.IsFreshTask(ctx)
```

Comment:
So we need to judge in every Tick?

Reply:
yes, until we set `needExtStorage` to `false`